flyteadmin/pkg/manager/impl/execution_manager.go

```diff
@@ -909,6 +917,14 @@ func (m *ExecutionManager) launchExecutionAndPrepareModel(
 		return nil, nil, nil, err
 	}
 
+	// If a parentNode launched this execution and it's a launchPlan, respect the maxParallelism set in the launchPlan.
```

Comment:
`and it's a launchPlan` can you elaborate here? Is this supposed to check the node type of the parent node?

IIUC this code is nullifying the overridden max parallelism configuration in the `requestSpec` if the execution has a `parentNodeExecutionID`.